addons/planner/static/src/js/planner.js

```diff
@@ -96,6 +97,8 @@
             'click .oe_planner div[id^="planner_page"] a[href^="#planner_page"]': 'change_page',
             'click .oe_planner li a[href^="#planner_page"]': 'change_page',
             'click .oe_planner div[id^="planner_page"] button[data-pageid^="planner_page"]': 'mark_as_done',
+            'hidden.bs.modal': 'modal_close',
+            'click .oe_planner .planner_link': 'save_input_value'
```

Comment:
I would bind events such as : 
* shown.bs.modal --> on_modal_show : bind an event `$(window).on("unload", this.save_input);`
* hidden.bs.modal --> on_modal_hide : unbind the event `$(window).off("unload", this.save_input);` + call `save_input`
When the modal is shown, if you click on an external link, the data will be saved. I'm not 100% sure, but I think it can really works.
Don't add class on link ;)